libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/ManeuverProcessor.kt

```diff
@@ -189,10 +190,25 @@ internal object ManeuverProcessor {
                 ifNonNull(routeLeg?.steps()) { steps ->
                     val stepList = mutableListOf<StepIndexToManeuvers>()
                     for (stepIndex in 0..steps.lastIndex) {
-                        steps[stepIndex].bannerInstructions()?.let { bannerInstruction ->
+                        val stepIntersections = if (stepIndex == steps.lastIndex) {
+                            steps[stepIndex].intersections()
+                        } else {
+                            steps[stepIndex + 1].intersections()
+                        }
```

Comment:
Do we have tests for this logic? Trying to prevent runtime `ArrayIndexOutOfBoundsException`s.

Reply:
Its a private method, not sure how I can particularly test this logic.

Reply:
Based on the [conversation](https://mapbox.slack.com/archives/C8LQNA2R1/p1627428161172600) it doesn't look like there is a need for any test here.

Reply:
We should add a test or at least a comment as to why this logic is present here. It's not a change that we'll remember a reason to in a year's time.

Reply:
@abhishek1508 could you follow up with a ticket/PR?